lib/xenopsd.ml

```diff
@@ -137,9 +123,25 @@ let configure ?(specific_options=[]) ?(specific_essential_paths=[]) ?(specific_n
 (** set up the BISECT_FILE env var to point to a temp directory where
  *  the log files go
  *)
-let setup_bisect () =
+let setup_bisect name =
   let (//) = Filename.concat in
-  try 
+  let tmpdir =
+    let getenv n   = try Sys.getenv n with Not_found -> "" in
+    let dirs    = 
+      [ getenv "TMP"
+      ; getenv "TEMP"
+      ; "/tmp"
+      ; "/usr/tmp"
+      ; "/var/tmp"
+      ] in
+    let is_dir  = function 
+    | ""    -> false
+    | path  -> try Sys.is_directory path with Sys_error _ -> false
+    in try
+      List.find is_dir dirs
+    with
+      Not_found -> error "can't find temp directory %s" __LOC__; exit 1
+  in try 
     ignore (Sys.getenv "BISECT_FILE") 
   with Not_found ->
     Unix.putenv "BISECT_FILE" (tmpdir // Printf.sprintf "bisect-%s" name)
```

Comment:
Can't you replace this with something like:
```
try 
   ignore (Sys.getenv "BISECT_FILE")
with Not_found ->
   Unix.putenv "BISECT_FILE" (Filename.temp_file (Printf.sprintf "bisect-%s" name) "out")
```

At least, you can replace the `let tmpdir = ...` with Filename.get_temp_dir_name`.